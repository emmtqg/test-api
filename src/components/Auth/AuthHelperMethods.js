import decode from 'jwt-decode';

import config from '../Config';

// The apiCamelizedLookup provides a fieldname check and crossreference
// for the form field names. It aids in translation of valid property
// names to whatever the API required (single point of definition between
// the state properties and api names). It was assumed if the ESLint rules
// did not allow underscores (as specified in the API names) then, rather
// than overriding the linter, a mechanism should be incorporated to allow
// any API to specify properties as required and they can then be updated
// in one location. If form fields are checked for testing, then the form
// component can also utilize the apiCamelizedLookup for field specifications.
const { apiCamelizedLookup } = config;
const apiNames = Object.keys(apiCamelizedLookup);

export default class AuthHelperMethods {
  getApiPropName = (name) => {
    // Get the API 'expected' property of input form field
    let index = -1;
    Object.values(apiCamelizedLookup).findIndex((ele, i) => {
      index = i;
      return (ele === name);
    });
    return index; // returning matched index or -1 if not in apiCamelized object
  }

  lookupApiName = (name) => {
    const index = this.getApiPropName(name);
    if (index > -1) {
      // apiNames are the API keys in lookup array
      return apiNames[index];
    }

    return ''; // API will error with invalid fieldname
  }

  register = (registerObject) => {
    const {
      firstName,
      lastName,
      email,
      password,
      confirmPassword,
    } = registerObject;

    // Get a token from api server using the fetch api
    return (
      this.fetch(config.registerUrl, {
        method: 'POST',
        body: JSON.stringify({
          [this.lookupApiName('firstName')]: firstName,
          [this.lookupApiName('lastName')]: lastName,
          email,
          password,
          [this.lookupApiName('confirmPassword')]: confirmPassword,
        }),
      })
        .then((response) => {
          // console.log(`Auth (Register): ${response}`);
          if (response) {
            this.setToken(response.token); // Setting the token in localStorage
            return Promise.resolve(response);
          }
          return Promise.reject(response);
        })
    );
  }

  newPlayer = (newPlayerObject) => {
    const {
      firstName,
      lastName,
      rating,
      handedness,
    } = newPlayerObject;

    // Get a token from api server using the fetch api
    return (
      this.fetch(config.playersUrl, {
        method: 'POST',
        body: JSON.stringify({
          [this.lookupApiName('firstName')]: firstName,
          [this.lookupApiName('lastName')]: lastName,
          rating,
          handedness,
        }),
      })
    );
  }

  getPlayers = () => (
    this.fetch(config.playersUrl, {
      method: 'GET',
    })
  );

  removePlayer = (playerId) => {
    const deleteUrl = `${config.playersUrl}/${playerId}`;
    return (
      this.fetch(deleteUrl, {
        method: 'DELETE',
      })
    );
  }

  login = (loginObject) => {
    const {
      email,
      password,
    } = loginObject;

    // Get a token from api server using the fetch api
    return (
      this.fetch(config.loginUrl, {
        method: 'POST',
        body: JSON.stringify({
          email,
          password,
        }),
      })
        .then((response) => {
          if (response) {
            // console.log(`Auth (Login): ${response}`);
            this.setToken(response.token); // Setting the token in localStorage
            return response;
          }
          return Promise.reject(response);
        })
        .catch((error) => {
          if (error && error.message) {
            // console.log(`Error: ${error.message}`);
          }
          return Promise.reject(error);
        })
    );
  }

  loggedIn = () => {
    // Checks if there is a saved token and it's still valid
    const token = this.getToken(); // Getting token from localstorage
    return (typeof token !== typeof undefined) && (!!token && !this.isTokenExpired(token));
  }

  isTokenExpired = (token) => {
    try {
      const decoded = decode(token);
      if (decoded.exp < Date.now() / 1000) { // Checking if token is expired.
        return true;
      }
      return false;
    } catch (err) {
      // console.log('Check for expired tolen failed.');
      return false;
    }
  }

  setToken = (idToken) => {
    // Saves user token to localStorage
    localStorage.setItem('id_token', idToken);
  }

  getToken = () =>
    // Retrieves the user token from localStorage
    localStorage.getItem('id_token')


  logout = () => {
    // Clear user token and profile data from localStorage
    localStorage.removeItem('id_token');
  }

  getConfirm = () => {
    // Using jwt-decode npm package to decode the token
    const answer = decode(this.getToken());
    // console.log(`Get confirm token => ${answer}`);
    return answer;
  }

  fetch = (url, options) => {
    // performs api calls sending the required authentication headers
    const headers = {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    };

    // Setting Authorization header
    // Authorization: Bearer xxxxxxx.xxxxxxxx.xxxxxx
    if (this.loggedIn()) {
      headers.Authorization = `Bearer ${this.getToken()}`;
    }

    return fetch(url, {
      headers,
      ...options,
    })
      .then(this.checkStatus)
      .then(response => response.json())
      .catch((error) => {
        // console.log(`Error: ${error.message}`);
        throw error;
      });
  }

  checkStatus = (response) => {
    // Success status lies between 200 to 300
    if (response.status >= 200 && response.status < 300) {
      return response;
    }

    // raises an error in case response status is not a success
    const error = new Error(response.statusText);
    error.response = response;
    error.status = response.status;
    throw error;
  }
}
