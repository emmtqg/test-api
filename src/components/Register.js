import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';

import config from './Config';
import './SideBySideComponents.css';

const styles = theme => ({
  main: {
    maxWidth: 800,
    width: 'auto',
    display: 'block', // Fix IE 11 issue.
    [theme.breakpoints.down('sm')]: {
      marginLeft: 5,
      marginRight: 5,
    },
    [theme.breakpoints.up((400 + theme.spacing.unit) * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`,
  },
  avatar: {
    margin: theme.spacing.unit,
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing.unit,
  },
  submit: {
    marginTop: theme.spacing.unit * 3,
  },
});

const { apiCamelizedLookup } = config;
class Register extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstName: '',
      lastName: '',
      password: '',
      confirmPassword: '',
      remember: false,
      isLoading: false,
      error: false,
      message: '',
      messageSuggest: '',
    };

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  componentWillMount() {
    // Here is a great place to redirect someone who is already logged in
    if (this.props.auth.loggedIn()) {
      this.props.history.replace('/');
    }
  }

  handleChange = (event) => {
    // 'normalize' the underscore id's with the
    // state property fields to both satify project
    // API requirements and ESLINT rules.
    const { name } = event.target;
    const stateName = apiCamelizedLookup[name];
    this.setState({
      [stateName]: event.target.value,
    });
  }

  handleFormSubmit = (e) => {
    e.preventDefault();

    /* confirm that password and password confirm are the same */
    if (!Object.is(this.state.password, this.state.confirmPassword)) {
      this.setState({
        message: 'Your confirmed password does not match the original password.',
        messageSuggest: 'Please recheck your password entry.',
      });
      return null;
    }

    // Here is where all the login logic will go. Upon clicking the login
    // button, a login method is invoked that will send the entered
    // credentials over to the server for verification. Once verified, it should
    // store the user's token and send you to the protected route.
    // this.props.auth.register(this.state);
    this.props.auth.register(this.state)
      .then(() => {
        this.setState({
          message: '',
          messageSuggest: '',
          isLoading: false,
        });

        return this.props.history.replace('/');
      })
      .catch((error) => {
        this.setState({
          message: `Registration encountered an error: (${error.status}) ${error.message}.`,
          messageSuggest: 'Please recheck your information.',
        });
      });

    return null;
  }

  render() {
    const { classes } = this.props;
    const { message } = this.state;

    return (
      <main className={classes.main}>
        <CssBaseline />
        <Paper className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Register
          </Typography>
          { (this.state.message) ? <div><br /><hr /></div> : null }
          <Typography style={{ color: 'red' }} component="h5" variant="h5">
            {message}<br />
            {this.state.messageSuggest}
          </Typography>
          <form id="Register" className={classes.form} onSubmit={this.handleFormSubmit}>
            <div className="flexGrid">
              <div className="col">
                <FormControl margin="normal" required className="almostWidth">
                  <InputLabel htmlFor="first_name">First Name</InputLabel>
                  <Input id="first_name" name="first_name" autoComplete="first_name" autoFocus onChange={this.handleChange} />
                </FormControl>
              </div>
              <div className="col">
                <FormControl margin="normal" required className="justifyEnd">
                  <InputLabel htmlFor="last_name">Last Name</InputLabel>
                  <Input id="last_name" name="last_name" autoComplete="last_name" onChange={this.handleChange} />
                </FormControl>
              </div>
            </div>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="email">Email Address</InputLabel>
              <Input id="email" name="email" autoComplete="email" onChange={this.handleChange} />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="password">Password</InputLabel>
              <Input name="password" type="password" id="password" autoComplete="current-password" onChange={this.handleChange} />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="confirm_password">Confirm Password</InputLabel>
              <Input name="confirm_password" type="password" id="confirm_password" onChange={this.handleChange} />
            </FormControl>
            <Button
              id="btnRegister"
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onSubmit={this.handleSubmit}
            >
              Register Me!
            </Button>
          </form>
        </Paper>
      </main>
    );
  }
}

Register.defaultProps = {
  auth: {},
  // History object passed via Router or a function
  // that passed in unit tests via jest.
  history: PropTypes.oneOfType([
    PropTypes.shape({
      replace: () => {},
    }),
    () => {},
  ]),
};

Register.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  classes: PropTypes.object.isRequired,
  auth: PropTypes.shape({
    loggedIn: PropTypes.func.isRequired,
    register: PropTypes.func.isRequired,
  }),
  // History object passed via Router or a function
  // that passed in unit tests via jest.
  history: PropTypes.oneOfType([
    PropTypes.shape({
      replace: PropTypes.func,
    }),
    PropTypes.func,
  ]),
};

export default withStyles(styles)(Register);
