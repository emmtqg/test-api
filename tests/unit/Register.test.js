import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import Register from '../../src/components/pages/Register';
import AuthHelperMethods from '../../src/components/auth/AuthHelperMethods';

// All field forms are translated between the form and API property
// names and the component state properties. This is defined in the
// project Config.js file.
const apiCamelizedLookupForRegister = {
  first_name: 'firstName',
  last_name: 'lastName',
  confirm_password: 'confirmPassword',
  email: 'email',
  password: 'password',
};

let UnwrappedShallowComponent;
let wrapper;
const history = {
  replace: jest.fn(),
};
const auth = new AuthHelperMethods();

// auth mocks for Promise return of newUser API call from AuthHelpers.js
const newUser = {
  success: true,
  token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1Y2E3NzI0YTMxMGQyMTdmZjM5MDFlYWUiLCJpYXQiOjE1NTQ0Nzc2NDJ9.dB--qqJQT6Zx8Ccn6HhKAXqtZX0oqhgyV2oEjy3RonU',
  user: {
    first_name: 'Same',
    last_name: 'Smith',
    email: 'a@b.com',
    id: '5ca7724a310d217ff3901eae',
  },
};
const registerMock = jest.fn().mockReturnValue(Promise.resolve(newUser));
auth.register = registerMock;

beforeAll(() => {
  wrapper = mount(<Register auth={auth} props={history} />);
  const shallowComponent = shallow(<Register auth={auth} props={history} />);
  UnwrappedShallowComponent = shallowComponent.first().shallow();
});

afterAll(() => {
  wrapper.unmount();
});

describe('The Register Form should', () => {
  it('match the Registration form snapshot', () => {
    expect(wrapper.find('Register').length).toEqual(1);
  });

  it('render correctly', () => {
    const component = renderer.create(<Register auth={auth} props={history} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('actually have a form element', () => {
    expect(wrapper.find('form').length).toBe(1);
  });

  // Loop through the apiCamelizedLookup array for all field forms
  describe('The form fields retrieve data and update component state', () => {
    // state names are transformed to camelCase for eslint rule
    // This is done via an enum like object defined in the project
    // Config.js file (under /src).
    const formNames = Object.keys(apiCamelizedLookupForRegister);
    const camelizedNames = Object.values(apiCamelizedLookupForRegister);
    let input;
    formNames.forEach((element) => {
      it(` checking ${element} has a defined onChange function`, () => {
        // MUI returning multiple nodes with #text_field, get the first with props
        input = wrapper.find(`#${element}`).first();
        expect(input.props('onChange')).toBeDefined();
        expect(typeof input.props('onChange').onChange === 'function').toBe(true);
      });
    });

    formNames.forEach((element, index) => {
      it(` checking ${element} updates state property on change`, () => {
        input = UnwrappedShallowComponent.find(`#${element}`).first();
        input.simulate('change', { target: { name: element, value: 'Ping' } });
        expect(UnwrappedShallowComponent.state(camelizedNames[index])).toBe('Ping');
      });
    });

    describe('The "Register Me!" button should', () => {
      const spy = sinon.spy();
      let pWrapper;
      beforeEach(() => {
        Register.prototype.handleFormSubmit = spy;
        pWrapper = mount(<Register auth={auth} />);
      });

      afterEach(() => {
        pWrapper.unmount();
      });

      it('process submit event for Register Me', () => {
        // fill in the form
        formNames.forEach((element) => {
          input = UnwrappedShallowComponent.find(`#${element}`).first();
          input.simulate('change', { target: { name: element, value: 'Ping' } });
        });

        // make confirm password differ to bump out of handleFormSubmit
        input = UnwrappedShallowComponent.find('#confirm_password').first();
        input.simulate('change', { target: { name: 'confirm_password', value: 'abcd' } });

        let prevented = false;
        UnwrappedShallowComponent.find('form').simulate('submit', { preventDefault: () => { prevented = true; } });
        expect(prevented).toBe(true);
      });

      it('do client-side check that password and confirm_password match testing state (stubbed)', () => {
        const stubHandleSubmit = sinon.stub(UnwrappedShallowComponent.instance(), 'handleFormSubmit').callsFake((thisComponent) => {
          const thisComponentState = thisComponent.state();
          return (thisComponentState.confirmPassword === thisComponentState.password);
        });

        // fill in the form
        formNames.forEach((element) => {
          input = UnwrappedShallowComponent.find(`#${element}`).first();
          input.simulate('change', { target: { name: element, value: 'Ping' } });
        });

        UnwrappedShallowComponent.find('form').simulate('submit', UnwrappedShallowComponent);
        sinon.assert.called(stubHandleSubmit);
      });

      it('Redirects to Roster (main page with Auth) on successful registration', () => {
        // mock history.replace('/roster')
        const mockReplaceURLtoRoster = jest.fn(() => true);
        const mockEvent = { preventDefault: () => {} };
        history.replace = mockReplaceURLtoRoster;

        const wrap = shallow(<Register auth={auth} props={history} />);
        const unwrappedWrap = wrap.first().shallow();

        // fill in the form
        input = unwrappedWrap.find('#first_name').first();
        input.simulate('change', { target: { name: 'first_name', value: 'Same' } });
        input = unwrappedWrap.find('#last_name').first();
        input.simulate('change', { target: { name: 'last_name', value: 'Smith' } });
        input = unwrappedWrap.find('#email').first();
        input.simulate('change', { target: { name: 'email', value: 'a@b.com' } });
        input = unwrappedWrap.find('#password').first();
        input.simulate('change', { target: { name: 'password', value: 'foozball' } });
        input = unwrappedWrap.find('#confirm_password').first();
        input.simulate('change', { target: { name: 'confirm_password', value: 'foozball' } });

        // Call submit and mock Auth.register to return resolved promise
        unwrappedWrap.find('form').simulate('submit', mockEvent);
        expect(registerMock('auth.register')).resolves.toBe(newUser);
      });
    });
  });
});
