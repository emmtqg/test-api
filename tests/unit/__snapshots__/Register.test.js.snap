// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Register Form should render correctly 1`] = `
<main
  className="Register-main-1"
>
  <div
    className="MuiPaper-root-6 MuiPaper-elevation2-10 MuiPaper-rounded-7 Register-paper-2"
  >
    <div
      className="MuiAvatar-root-33 MuiAvatar-colorDefault-34 Register-avatar-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-36"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0V0z"
          />
          <path
            d="M0 0h24v24H0V0z"
            opacity=".87"
          />
        </g>
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root-45 MuiTypography-h5-61"
    >
      Register
    </h1>
    <h5
      className="MuiTypography-root-45 MuiTypography-h5-61"
      style={
        Object {
          "color": "red",
        }
      }
    >
      
      <br />
      
    </h5>
    <form
      className="Register-form-4"
      id="Register"
      onSubmit={[Function]}
    >
      <div
        className="flexGrid"
      >
        <div
          className="col"
        >
          <div
            className="MuiFormControl-root-81 MuiFormControl-marginNormal-82 almostWidth"
          >
            <label
              className="MuiFormLabel-root-96 MuiFormLabel-required-101 MuiInputLabel-required-89 MuiInputLabel-root-85 MuiInputLabel-formControl-90 MuiInputLabel-animated-93"
              data-shrink={false}
              htmlFor="first_name"
            >
              First Name
              <span
                className="MuiFormLabel-asterisk-102"
              >
                 *
              </span>
            </label>
            <div
              className="MuiInputBase-root-116 MuiInput-root-103 MuiInput-underline-107 MuiInputBase-formControl-117 MuiInput-formControl-104"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="first_name"
                autoFocus={true}
                className="MuiInputBase-input-126 MuiInput-input-111"
                disabled={false}
                id="first_name"
                name="first_name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="MuiFormControl-root-81 MuiFormControl-marginNormal-82 justifyEnd"
          >
            <label
              className="MuiFormLabel-root-96 MuiFormLabel-required-101 MuiInputLabel-required-89 MuiInputLabel-root-85 MuiInputLabel-formControl-90 MuiInputLabel-animated-93"
              data-shrink={false}
              htmlFor="last_name"
            >
              Last Name
              <span
                className="MuiFormLabel-asterisk-102"
              >
                 *
              </span>
            </label>
            <div
              className="MuiInputBase-root-116 MuiInput-root-103 MuiInput-underline-107 MuiInputBase-formControl-117 MuiInput-formControl-104"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="last_name"
                className="MuiInputBase-input-126 MuiInput-input-111"
                disabled={false}
                id="last_name"
                name="last_name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root-81 MuiFormControl-marginNormal-82 MuiFormControl-fullWidth-84"
      >
        <label
          className="MuiFormLabel-root-96 MuiFormLabel-required-101 MuiInputLabel-required-89 MuiInputLabel-root-85 MuiInputLabel-formControl-90 MuiInputLabel-animated-93"
          data-shrink={false}
          htmlFor="email"
        >
          Email Address
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-116 MuiInput-root-103 MuiInput-underline-107 MuiInputBase-formControl-117 MuiInput-formControl-104"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            className="MuiInputBase-input-126 MuiInput-input-111"
            disabled={false}
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-81 MuiFormControl-marginNormal-82 MuiFormControl-fullWidth-84"
      >
        <label
          className="MuiFormLabel-root-96 MuiFormLabel-required-101 MuiInputLabel-required-89 MuiInputLabel-root-85 MuiInputLabel-formControl-90 MuiInputLabel-animated-93"
          data-shrink={false}
          htmlFor="password"
        >
          Password
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-116 MuiInput-root-103 MuiInput-underline-107 MuiInputBase-formControl-117 MuiInput-formControl-104"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            className="MuiInputBase-input-126 MuiInput-input-111 MuiInputBase-inputType-129 MuiInput-inputType-114"
            disabled={false}
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-81 MuiFormControl-marginNormal-82 MuiFormControl-fullWidth-84"
      >
        <label
          className="MuiFormLabel-root-96 MuiFormLabel-required-101 MuiInputLabel-required-89 MuiInputLabel-root-85 MuiInputLabel-formControl-90 MuiInputLabel-animated-93"
          data-shrink={false}
          htmlFor="confirm_password"
        >
          Confirm Password
          <span
            className="MuiFormLabel-asterisk-102"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-116 MuiInput-root-103 MuiInput-underline-107 MuiInputBase-formControl-117 MuiInput-formControl-104"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-126 MuiInput-input-111 MuiInputBase-inputType-129 MuiInput-inputType-114"
            disabled={false}
            id="confirm_password"
            name="confirm_password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-159 MuiButton-root-133 MuiButton-contained-144 MuiButton-containedPrimary-145 MuiButton-raised-147 MuiButton-raisedPrimary-148 MuiButton-fullWidth-158 Register-submit-5"
        disabled={false}
        id="btnRegister"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-134"
        >
          Register Me!
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
    </form>
  </div>
</main>
`;
